modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/loader/java/type/runtime/TypeWrapperTestCase.java

```diff
@@ -63,6 +65,16 @@ public void filterInternalJdkFields() {
     assertThat(fields, empty());
   }
 
+  @Test
+  @Issue("W-13648907")
+  @Description("Ensure synthetic fields are filtered out as they might be added to classes we can't use reflection on in Java 17 and above")
+  public void filterSyntheticFields() {
+    TypeWrapper type = new TypeWrapper(SomeClass.class, new DefaultExtensionsTypeLoaderFactory()
```

Comment:
SomeClass doesn't look like it would have synthetic fields in the first place. I think the test should start by asserting that SomeClass actually does have synthetic fields

Reply:
it's a non-static inner class, it has the `this$0` field. will add it to the test